torch/_dynamo/output_graph.py

```diff
@@ -533,7 +535,12 @@ def compile_subgraph(
         if stack_values:
             val_to_names[stack_values[-1]] = list()
         for k, v in tx.symbolic_locals.items():
-            if isinstance(v.source, LocalSource) and v.source.name() == k:
+            # Note! this explicitly uses .local_name for matching
+            # Failure to do so will cause spurious registrations in val_to_names.
+            # This will in turn result in spurious variables showing up in the graph.
+            # This was very tricky to debug. For an example, dump the graph at call_user_compiler
+            # while running test_subgraphs.py
+            if isinstance(v.source, LocalSource) and v.source.local_name == k:
```

Comment:
I think it is a lot clearer if we do `v.source == LocalSource(k)` (this is assuming LocalSource has no subclasses)

Reply:
LocalSource has LocalInputSource subclass, but I can clean that up actually, we don't really need it anymore.

Reply:
Done